test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextNoTrackingStreamsTests.cs

```diff
@@ -0,0 +1,450 @@
+                new DataServiceContext[] { NonTrackingContext, DefaultTrackingContext }, false);
+
+            var user = new User
+            {
+                Name = "Some name"
+            };
+
+            DefaultTrackingContext.AddObject("Users", user);
+            NonTrackingContext.AddObject("Users", user);
+            Assert.NotNull(NonTrackingContext.GetEntityDescriptor(user));
+            Assert.NotNull(DefaultTrackingContext.GetEntityDescriptor(user));
+
+            SaveContextChanges(new DataServiceContext[] { DefaultTrackingContext, NonTrackingContext });
+            Assert.Equal(1, DefaultTrackingContext.Entities.Count);
+            Assert.Equal(1, NonTrackingContext.Entities.Count);
+        }
+
+        private void SaveContextChanges(DataServiceContext[] dataServiceContexts)
+        {
+            foreach (var dataServiceContext in dataServiceContexts)
+            {
+                dataServiceContext.SaveChanges();
+            }
+        }
+#else
+ [Fact]
+        public async Task TestAddingNewItemsBehaviourShouldBeUnAlteredAsync()
+        {
+            SetupContextWithRequestPipelineForSaving(
+                new DataServiceContext[] { NonTrackingContext, DefaultTrackingContext }, false);
+
+            var user = new User
+            {
+                Name = "Some name"
+            };
+
+            DefaultTrackingContext.AddObject("Users", user);
+            NonTrackingContext.AddObject("Users", user);
+            Assert.NotNull(NonTrackingContext.GetEntityDescriptor(user));
+            Assert.NotNull(DefaultTrackingContext.GetEntityDescriptor(user));
+
+            await SaveContextChangesAsync(new DataServiceContext[] { DefaultTrackingContext, NonTrackingContext });
+            Assert.Equal(1, DefaultTrackingContext.Entities.ToList().Count);
+            Assert.Equal(1, NonTrackingContext.Entities.ToList().Count);
+        }
+
+#endif
+
+
+        [Fact]
+        public async void GetNamedStreamsShouldWorkInNoTrackingModeIfExtendsBaseEntity()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, true, true);
+
+            var customers = await NonTrackingContext.Customers.ExecuteAsync();
+
+            var photoUri = NonTrackingContext.GetReadStreamUri(customers.First(), "ProfilePhoto");
+            Assert.NotNull(photoUri);
+            Assert.Equal("https://localhost:8000/Customers(1)/ProfilePhoto", photoUri.ToString());
+        }
+
+        [Fact]
+        public async Task GetNamedStreamsShouldNotWorkInNoTrackingModeIfNotExtendsBaseEntity()
+        {
+            SetupContextWithRequestPipeline(new DataServiceContext[] { NonTrackingContext }, true, true);
+
+            var customers = await NonTrackingContext.Customers2.ExecuteAsync();
+
+            Action action = () => NonTrackingContext.GetReadStreamUri(customers.First(), "ProfilePhoto");
+            action.ShouldThrow<InvalidOperationException>();
+        }

```

Comment:
Why should it not work without extending `BaseEntity`?

Reply:
I think using base entity provides a way to opt-in to the behavior. Once the new changes are done to have these in an interface then this could be refactored a bit to introduce some interface for this and document. probably this would be a great candidate for the new version 8.0 👍